starlite/asgi/routing_trie/traversal.py

```diff
@@ -91,7 +91,7 @@ def traverse_route_map(
     has_path_param = PathParameterSentinel in current_node["child_keys"]
 
     if not path_components:
-        if has_path_param or not current_node["asgi_handlers"]:
+        if not current_node["asgi_handlers"]:
```

Comment:
I don't think that it matters at this point if a child node has a path parameter or not, if there are no components left and the current node has handlers it shouldn't be a 404.

Sound right?